src/server.rs

```diff
@@ -589,7 +633,21 @@ impl RollupBoostServer {
                 info!(message = "using flashblocks payload");
                 OpExecutionPayloadEnvelope::V3(payload)
             } else {
-                builder.get_payload(payload_id, version).await?
+                match builder.get_payload(payload_id, version).await {
+                    Ok(payload) => payload,
+                    Err(e) => {
+                        error!(message = "error getting builder payload", "error" = %e);
+                        match e {
```

Comment:
Couldn't find if this will return the expected status code from the `rbuilder`, any pointers on determining this?